cmake/create_test.cmake

```diff
@@ -87,9 +151,9 @@ endfunction()
 function(ginkgo_create_test test_name)
     ginkgo_build_test_name(${test_name} test_target_name)
     add_executable(${test_target_name} ${test_name}.cpp)
-    target_link_libraries(${test_target_name} PRIVATE ${create_test_ADDITIONAL_LIBRARIES})
+    target_link_libraries(${test_target_name})
     ginkgo_set_test_target_properties(${test_target_name} ${ARGN})
-    ginkgo_add_test(${test_name} ${test_target_name} ${ARGN})
+    ginkgo_add_test(${test_name} ${test_target_name} ${ARGN} RESOURCE_TYPE ref)
 endfunction(ginkgo_create_test)
```

Comment:
if this allows RESOURCE_TYPE input, the re-naming and changing test macro in CMakeFile is not neccessary.
like `ginkgo_create_test(memory RESOURCE_TYPE CUDA)` not `ginkgo_create_cuda_test(memory)` with linking additional library

Reply:
The renaming seems cleaner to me, since it handles all tests uniformly. Otherwise, why are some tests just `ginkgo_create_test` while others are `ginkgo_create_cuda_test` when both are testing cuda functionality.

Reply:
ginkgo_create_cuda_test means the file contains some cuda functionalities and needs nvcc compiler to compile.
ginkgo_create_test only uses host compiler.
If it only tests the functionality without requiring nvcc, it should use host compiler